### PR TITLE
Integrate meta-lhg into i.MX8M

### PR DESCRIPTION
**Pull request by ajutras-linaro**

### CODE REVIEW

File: meta-lhg/recipes-browser/chromium/chromium-wayland_%.bbappend [hunk 1]
```diff
@@ -1,4 +1,6 @@
 PACKAGECONFIG_append = " ${@bb.utils.contains('MACHINE_FEATURES', 'optee', 'use-ocdm', '', d)}  proprietary-codecs"
+PACKAGECONFIG[use-ocdm] = ",,,"
```
Comment (andrey-konovalov) [inline meta-lhg/recipes-browser/chromium/chromium-wayland_%.bbappend]: Just 
PACKAGECONFIG[use-ocdm] = ""
is enough

Comment (ajutras-linaro) [reply meta-lhg/recipes-browser/chromium/chromium-wayland_%.bbappend]: I updated the code base on this comment.

**Comment by petegriffin** [comment]
LGTM

**Comment by ajutras-linaro** [comment]
Thank you Andrey and Peter for the review.

**Comment by andrey-konovalov** [comment]
Actually I am not quite happy with the SOC-specific override (OPTEE_EXPORT_USERTA_DIR_mx8mq) in meta-lhg layer (meta-lhg/meta-lhg/ that is). Going to take a look at how optee-os is handled - maybe there is something there that we could reuse to keep meta-lhg layer hardware-agnostic...

**Comment by ajutras-linaro** [comment]
I agree it would be better to have the same OP-TEE user TA path for all devices. I suggest to use the current patch temporarily and that we improve the solution when the integration is more mature.

File: meta-lhg/recipes-security/optee/optee-aes-decryptor.bb [hunk 2]
```diff
@@ -16,12 +16,19 @@ SRCREV = "${AUTOREV}"
 S = "${WORKDIR}/git"
 B = "${S}"
 
+OPTEE_ARCH ?= "arm32"
+OPTEE_ARCH_armv7a = "arm32"
+OPTEE_ARCH_aarch64 = "arm64"
+
+OPTEE_EXPORT_USERTA_DIR ?= "export-user_ta"
+OPTEE_EXPORT_USERTA_DIR_mx8mq = "export-user_ta_${OPTEE_ARCH}"
```
Comment (psivasubramanian) [inline meta-lhg/recipes-security/optee/optee-aes-decryptor.bb:24]: Can this be moved to iMX layer?

File: meta-lhg/recipes-security/optee/optee-aes-decryptor.bb [hunk 3]
```diff
@@ -16,12 +16,19 @@ SRCREV = "${AUTOREV}"
 S = "${WORKDIR}/git"
 B = "${S}"
 
+OPTEE_ARCH ?= "arm32"
+OPTEE_ARCH_armv7a = "arm32"
+OPTEE_ARCH_aarch64 = "arm64"
```
Comment (psivasubramanian) [inline meta-lhg/recipes-security/optee/optee-aes-decryptor.bb:21]: bb contains will be more ideal for this, I feel.

Comment (ajutras-linaro) [reply meta-lhg/recipes-security/optee/optee-aes-decryptor.bb:21]: Once the integration is completed, Andrey will look at how OP-TEE is integrated for i.MX8M. He will suggest some improvements.

**Pull request merged**